### PR TITLE
🔨 Simplify generic variants, update DFU tool

### PR DESCRIPTION
**Pull request by thinkyhead**
- Update to DFU tool 1.11.0 with minor patches — for better macOS compatibility (eventually). See [this thread](https://github.com/platformio/platform-ststm32/issues/436#issuecomment-2442483054).
- Modify "variant" scripts to point the `build.variants_dir` at our folder (instead of copying into the platform).
- Rename some boards and variants to match siblings.

### CODE REVIEW

File: buildroot/share/PlatformIO/scripts/STM32F1_create_variant.py [hunk 1]
```diff
@@ -3,28 +3,40 @@
...
+    board = env.BoardConfig()
     variant = board.get("build.variant")
-    variant_dir = FRAMEWORK_DIR / "STM32F1/variants" / variant
 
     source_dir = source_root / variant
     assert source_dir.is_dir()
 
-    if variant_dir.is_dir():
-        shutil.rmtree(variant_dir)
+    if True:
```
Comment (thijstriemstra) [inline buildroot/share/PlatformIO/scripts/STM32F1_create_variant.py:19]: ?

Comment (thinkyhead) [reply buildroot/share/PlatformIO/scripts/STM32F1_create_variant.py:19]: Developers always leaving placeholders, amirite?

**Comment by thinkyhead** [comment]
Double-checked to make sure files are indeed being included, and it all seems good. Maple can't take the same approach, but maybe it's possible to set extra environment vars or other build flags to sort out `#include <...>` lines within the Maple variants.

File: buildroot/share/PlatformIO/scripts/STM32F1_create_variant.py (hunk 1 above)
Comment (thijstriemstra) [reply buildroot/share/PlatformIO/scripts/STM32F1_create_variant.py:19]: fair enough, but i didnt expect it to be actually merged to main ;)

**Comment by vehystrix** [comment]
Since the change in `generic_create_variant.py` I get build errors. I get as far as this:
```
Linking .pio\build\MKS_E3_V2\firmware.elf
arm-none-eabi-g++: error: C:/Users/VEH/Documents/3DPrint/Printer: No such file or directory
arm-none-eabi-g++: error: upgrades/Marlin/buildroot/share/PlatformIO/variants/MARLIN_MKS_E3/ldscript.ld: No such file or directory
*** [.pio\build\MKS_E3_V2\firmware.elf] Error 1
```

Note that I am using the path `C:\Users\VEH\Documents\3DPrint\Printer upgrades\Marlin` as my marlin source path. When I remove the space in the path, the build completes as it should.

**Pull request merged**